tailscale/client.go

```diff
@@ -45,6 +47,25 @@ type (
 const baseURL = "https://api.tailscale.com"
 const contentType = "application/json"
 
+// NewClientFromOAuth returns a new instance of the Client type that will perform operations against a chosen tailnet using
+// OAuth Credentials. The Client will generate a new API key against the Tailscale OAuth token endpoint.
+// If the API key expires, it will not be automatically renewed by the Client.
+// Additional options can be provided, see ClientOption for more details.
+func NewClientFromOAuth(oauthClientID, oauthClientSecret, tailnet string, options ...ClientOption) (*Client, error) {
```

Comment:
Could you please make this accept a `ctx.Context` instead of using `context.Background`?